source/NVDAObjects/IAccessible/ia2Web.py

```diff
@@ -213,8 +214,27 @@ def _get_states(self):
 			if popupState:
 				states.discard(controlTypes.State.HASPOPUP)
 				states.add(popupState)
+		if config.conf["documentFormatting"]["reportLinkType"] and self.isInternalLink:
+			states.add(controlTypes.State.INTERNAL_LINK)
 		return states
 
+	@property
+	def isInternalLink(self) -> bool:
+		if self.role != controlTypes.Role.LINK:
+			return False
+		value = self.value
+		if not value:
+			return False
+		if not hasattr(self, "treeInterceptor"):
+			return False
+		ti = self.treeInterceptor
+		if ti is None or not hasattr(ti, "documentConstantIdentifier"):
+			return False
```

Comment:
I think this can be shortened a bit.

```suggestion
		ti = getattr(self, "treeInterceptor", None)
		if ti is None or not hasattr(ti, "documentConstantIdentifier"):
			return False
```